Microsoft.Toolkit.Uwp.UI.Controls/Graph/SharePointFiles/SharePointFiles.cs

```diff
@@ -0,0 +1,558 @@
+    /// </summary>
+    public partial class SharePointFiles : Control
+    {
+        /// <summary>
+        /// File is selected
+        /// </summary>
+        public event EventHandler<FileSelectedEventArgs> FileSelected;
+
+        private GraphServiceClient _graphClient;
+        private string _driveId;
+        private Stack<string> _driveItemPath = new Stack<string>();
+        private IDriveItemChildrenCollectionRequest _nextPageRequest = null;
+        private CancellationTokenSource _cancelUpload = new CancellationTokenSource();
+        private CancellationTokenSource _cancelLoadFile = new CancellationTokenSource();
+        private CancellationTokenSource _cancelGetDetails = new CancellationTokenSource();
+
+        private ListView _list;
+        private Button _back;
+        private Button _upload;
+        private Button _share;
+        private Button _download;
+        private Button _delete;
+        private HyperlinkButton _error;
+        private Button _cancel;
+        private Button _hasMore;
+        private Grid _thumbnail;
+        private Windows.UI.Xaml.Controls.Image _thumbnailImage;
+        private ScrollViewer _details;
+        private TextBlock _status;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SharePointFiles"/> class.
+        /// </summary>
+        public SharePointFiles()
+        {
+            DefaultStyleKey = typeof(SharePointFiles);
+        }
+
+        private static async void GraphAccessTokenPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            SharePointFiles control = d as SharePointFiles;
+            control._graphClient = Common.GetAuthenticatedClient(control.GraphAccessToken);
+            if (!string.IsNullOrEmpty(control.DriveUrl))
+            {
+                await control.InitDrive(control.DriveUrl);
+            }
+        }
+
+        private static async void DriveUrlPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            SharePointFiles control = d as SharePointFiles;
+            if (control._graphClient != null && !string.IsNullOrWhiteSpace(control.DriveUrl))
+            {
+                if (Uri.IsWellFormedUriString(control.DriveUrl, UriKind.Absolute))
+                {
+                    await control.InitDrive(control.DriveUrl);
+                }
+            }
+        }
+
+        private static void DetailPanePropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            SharePointFiles control = d as SharePointFiles;
+            if (control.IsDetailPaneVisible)
+            {
+                control.ShowDetailsPane();
+            }
+        }
+
+        /// <summary>
+        /// Called when applying the control template.
+        /// </summary>
+        protected override void OnApplyTemplate()
+        {
+            _list = GetTemplateChild("list") as ListView;
```

Comment:
You shouldn't need to grab all these elements, especially if you set up more XAML for all your visibility toggling based on the properties of your class.

Reply:
done